### ISSUE TITLE
NSPhotoLibraryUsageDescription on iOS

### ISSUE THREAD

**Issue by sebastian-zarzycki-es** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#1079]
The documentation says "This plugin requires permissions to the users photos.". I was wondering - why, actually? It doesn't contain any functionality of users selecting photo from photo library or camera roll - the only option is to pass an URL. 

Is this permission needed, because the plugin still needs "read" permissions to access that photo library file? (file:/// and so on)

**Comment by jcesarmobile** [comment]
the plugin has a function to save to the photo album, so probably only used for that
https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin#saving-images-to-the-photo-album-ios-only-currently

**Comment by sebastian-zarzycki-es** [comment]
No. That's a separate permission -`NSPhotoLibraryAddUsageDescription`. And it makes sense that it uses it (because it does), for this function. But not `NSPhotoLibraryUsageDescription`.

**Comment by jcesarmobile** [comment]
`NSPhotoLibraryAddUsageDescription` was introduced on iOS 11 and it's used only to save, but before that `NSPhotoLibraryUsageDescription ` was used for both saving and reading on iOS 10 and older.

https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW73

**Comment by sebastian-zarzycki-es** [comment]
Ah. That makes sense. It would mean that `NSPhotoLibraryUsageDescription` is no longer needed for this plugin now.

**Comment by jcesarmobile** [comment]
Is needed if you support iOS 10 (cordova-ios 5.1.1 is the latest and still supports iOS 10)

**Comment by sebastian-zarzycki-es** [comment]
Well, yes, I guess. But, as a sidenote, in iOS world, the migration rate is fairly swift. < 12.x share is about 9% now (according to Apple) - while other data show it might be actually much less than(~4%) that in most cases (https://david-smith.org/iosversionstats/).
